sway/tree/layout.c

```diff
@@ -138,22 +140,46 @@ struct sway_container *container_remove_child(struct sway_container *child) {
 	return container_reap_empty(parent);
 }
 
-void container_move_to(struct sway_container* container,
-		struct sway_container* destination) {
+void container_move_to(struct sway_container *container,
+		struct sway_container *destination) {
 	if (container == destination
 			|| container_has_anscestor(container, destination)) {
 		return;
 	}
 	struct sway_container *old_parent = container_remove_child(container);
 	container->width = container->height = 0;
-	struct sway_container *new_parent =
-		container_add_sibling(destination, container);
+	struct sway_container *new_parent;
+	if (destination->type == C_VIEW) {
+		new_parent = container_add_sibling(destination, container);
+	} else {
+		new_parent = destination;
+		container_add_child(destination, container);
+	}
+	wl_signal_emit(&container->events.reparent, old_parent);
+	if (container->type == C_WORKSPACE) {
+		struct sway_seat *seat = sway_input_manager_get_default_seat(
+				input_manager);
```

Comment:
handler context has a seat.

Reply:
I don't want to bring the seat into this level of the code (for the time being)